src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,50 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    public static final int DEFAULT_CAPACITY = 10;
+    public static final int MULTIPLIER = 2;
+    private Object[] data;
+    private int capacity;
+    private int size;
+
+    public MateStack() {
+        this.data = new Object[DEFAULT_CAPACITY];
```

Comment:
```suggestion
        data = new Object[DEFAULT_CAPACITY];
```
you don't have shadowing variable here, this keyword is not necessary

Reply:
done